homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,247 @@
+    name="Network Interface",
+)
+
+
+# Utility function for setting up device info consistently
+def build_device_info(device, domain):
+    """Build device info helper function."""
+    return {
+        "identifiers": {(domain, device.get("ID", "Unknown"))},
+        "name": device.get("Name") or f"WatchYourLAN {device.get('ID', 'Unknown')}",
+        "manufacturer": device.get("Hw", "Unknown Manufacturer"),
+        "model": "WatchYourLAN Device",
+        "sw_version": device.get("Last_Seen", "Unknown"),
+    }
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator = entry.runtime_data["coordinator"]
+
+    # Define a list that can store multiple sensor types
+    entities: list[SensorEntity | CoordinatorEntity] = []
+
+    # Create device count sensor
+    entities.append(WatchYourLANDeviceCountSensor(coordinator))
+
+    # Loop over each device and create sensors for online status, IP, MAC, and Iface
+    for device in coordinator.data:
+        if isinstance(device, dict):
+            entities.append(
+                WatchYourLANSensor(coordinator, device)
+            )  # Online/Offline sensor
+            entities.append(WatchYourLANIPSensor(coordinator, device))  # IP sensor
+            entities.append(WatchYourLANMacSensor(coordinator, device))  # MAC sensor
+            entities.append(
+                WatchYourLANIfaceSensor(coordinator, device)
+            )  # Iface sensor
+
+    # Add all the entities
+    async_add_entities(entities)
+
+
+class WatchYourLANSensor(CoordinatorEntity, RestoreSensor, SensorEntity):
+    """Representation of a WatchYourLAN online/offline sensor."""
+
+    def __init__(
+        self, coordinator: WatchYourLANUpdateCoordinator, device: dict
+    ) -> None:
+        """Initialize the sensor for online/offline state."""
+        super().__init__(coordinator)
+        self.device = device
+        self.entity_description = ONLINE_STATUS_DESCRIPTION
+
+        # Set the unique ID and device info
+        self._attr_unique_id = f"{self.device.get('ID')}_online_status"
+        self._attr_device_info = build_device_info(self.device, DOMAIN)
+
+        # Set an appropriate icon for the sensor
+        self._attr_icon = (
+            "mdi:lan-connect"
+            if self.device.get("Now", 0) == 1
+            else "mdi:lan-disconnect"
+        )
+
+    @property
+    def native_value(self) -> str:
+        """Return the native value of the sensor."""
+        return "Online" if self.device.get("Now", 0) == 1 else "Offline"
+
+    async def async_update(self):
+        """Update the sensor's state from the coordinator."""
+        await (
+            self.coordinator.async_request_refresh()
+        )  # Request data update from the coordinator
+        self._attr_native_value = (
+            "Online" if self.device.get("Now", 0) == 1 else "Offline"
+        )
```

Comment:
The idea of entity_descriptions is that you create 1 generic base entity, and then use entity_descriptions to change how they look and what data to expose